src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -323,6 +348,58 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_annotate_from_pod_status() {
+        let cases = vec![
```

Comment:
Tests are missing some important test cases:
- non-default `FieldsSpec` with custom field names for `pod_ips` (and `pod_ip`)
- a test case where `PodStatus::pod_ips` is set, but the `PodStatus::pod_ip` is absent

Reply:
I'm not 100% certain on this second one. I don't think there should ever be a case where `PodStatus` includes`PodIPs` with entries but no `PodIP`. Perhaps I'm misunderstanding that though?

Reply:
This is correct - my idea is why don't we test this anyway. It might be the case that it'll be deprecated in the future or sth, so it is useful to have it covered. Plus, it's sort of a matrix of possibilities we test, and without this case, the coverage would be incomplete 😄

Reply:
cool sounds good then :+1: